src/winfspy/plumbing/security_descriptor.py

```diff
@@ -40,6 +40,26 @@ def from_string(cls, string_format):
             )
         return cls(psd[0], psd_size[0])
 
+    def to_string(self):
+        pwstr = ffi.new("PWSTR*")
+        pwstr_size = ffi.new("ULONG*")
+        flags = (
+            lib.WFSPY_OWNER_SECURITY_INFORMATION
+            | lib.WFSPY_GROUP_SECURITY_INFORMATION
+            | lib.WFSPY_DACL_SECURITY_INFORMATION
+            | lib.WFSPY_SACL_SECURITY_INFORMATION
+        )
+        if not lib.ConvertSecurityDescriptorToStringSecurityDescriptorW(
+            self.handle, lib.WFSPY_STRING_SECURITY_DESCRIPTOR_REVISION, flags, pwstr, pwstr_size
```

Comment:
[According to the documentation](https://docs.microsoft.com/en-us/windows/win32/api/sddl/nf-sddl-convertsecuritydescriptortostringsecuritydescriptorw) the `StringSecurityDescriptorLen` parameter can be set to NULL if don't need to retrieve the size of the data.

Reply:
Done! 8604a03